### DISCUSSION TITLE
GPU driver support

### DISCUSSION THREAD

**Discussion by Talador12** [renovatebot/renovate#29364] [Suggest an Idea]
### Tell us more.

Nvidia (CUDA) and AMD (ROCm) both have dependencies that are required to be updated periodically. These can be in the form of many different files and images.

Here are a few examples:
https://gitlab.com/nvidia/container-images/cuda/-/blob/master/dist/12.4.1/ubuntu2204/base/Dockerfile
https://gitlab.com/nvidia/container-images/cuda/-/blob/master/dist/12.4.1/ubuntu2204/runtime/Dockerfile

Maintaining these upgrades takes a decent amount of manual intervention to ensure every value is updated.

Would it be possible to add support for GPU driver upgrades with renovate?

I am certain you could also work with Nvidia and AMD directly on this, as they are very involved in the open source community to enhance their ecosystems.

**Comment by rarkins** [comment]
It looks feasible but we'd need more specifics about where the list of versions should be fetched from, etc.

**Reply by nerzhul** [reply]
I'm working on making it work on our golden images for our customers though renovate.json, i didn't succeeded yet, but here is the source for versions
https://docs.nvidia.com/datacenter/tesla/drivers/releases.json

**Reply by rarkins** [reply]
We now have custom datasource support which can probably be used to transform this into a Renovate-compatible datasource

**Reply by nerzhul** [reply]
@rarkins if you have something i'll take it. For now i used this json datasource to generate package names and it work as expected

**Reply by rarkins** [reply]
Hi, here is the custom json datasource: https://docs.renovatebot.com/modules/datasource/custom/
